### ISSUE TITLE
Relation search by id with MongoDB throw an error

### ISSUE THREAD

**Issue by aksdevac** [strapi/strapi#5950]
**Describe the bug**
Hello, I am getting following error while typing in search box of 1 relational field on model create/update page of strapi admin
```
GET /content-manager/explorer/application::{{modelName}}?id_contains=54&_limit=20&_start=0 (7 ms) 500
[2020-04-25T10:42:30.363Z] error Error: Can't use $options
    at ObjectId.SchemaType.castForQuery (/Users/{{username}}/work/projects/{{projectname}}/node_modules/mongoose/lib/schematype.js:1374:13)
    at ObjectId.SchemaType.castForQueryWrapper (/Users/{{username}}/work/{{projectname}}//{{projectname}}/node_modules/mongoose/lib/schematype.js:1353:17)
    at cast (/Users/{{username}}/work/projects/{{projectname}}/node_modules/mongoose/lib/cast.js:296:39)
    at cast (/Users/{{username}}/work/projects/{{projectname}}/node_modules/mongoose/lib/cast.js:58:18)
    at model.Query.Query.cast (/Users/{{username}}/work/projects/{{projectname}}/node_modules/mongoose/lib/query.js:4697:12)
    at model.Query.Query._castConditions (/Users/{{username}}/work/projects/{{projectname}}/node_modules/mongoose/lib/query.js:1876:10)
    at model.Query.<anonymous> (/Users/{{username}}/work/projects/{{projectname}}/node_modules/mongoose/lib/query.js:1903:8)
    at model.Query._wrappedThunk [as _find] (/Users/{{username}}/work/projects/{{projectname}}/node_modules/mongoose/lib/helpers/query/wrapThunk.js:16:8)
    at /Users/{{username}}/work/projects/{{projectname}}/node_modules/kareem/index.js:278:20
    at /Users/{{username}}/work/projects/{{projectname}}/node_modules/kareem/index.js:77:15
    at processTicksAndRejections (internal/process/task_queues.js:79:11)
```
**Steps to reproduce the behavior**

1. Go to create/edit page of model having relational field, in my case 'has many' relation but tried in dependant model so only single value can be selected from dropdown(with search by id/_id field, I configured view to display id (Entry title) in relational field ) 
2. Click on relational field dropdown & search for any id in create/edit page
3. It should show Error occrured message in UI & log error on server

**Expected behavior**
It is not breaking anything but displays error on UI & server log

**System**

- Node.js version: 12.16.2 
- NPM version: 6.14.4
- Strapi version: 3.0.0-beta.20 
- Database: MongoDB
- Operating system: MacOS 10.15.4 or Ubuntu 16.04

**Possible Cause**
As it only happens if I select id as field to display, so I think may be mongoose don't allow passing $options for ignore case for `_id` field

**Comment by lauriejim** [comment]
Hello! Thank you for reporting this issue.

**Comment by viters** [comment]
@lauriejim 
The problem is way more serious than just showing error. It effectively stops you from searching / attaching relations by ID via admin panel.

**Comment by viters** [comment]
Fast fix for people who need this to be working (like me):

Create file: app/extensions/content-manager/controllers/ContentManager.js

Reimplement find to use `id_eq` comparator instead of `id_contains`:
```js
'use strict';

const _ = require('lodash');

module.exports = {
  /**
   * Returns a list of entities of a content-type matching the query parameters
   */
  async find(ctx) {
    const { model } = ctx.params;
    const contentManagerService =
      strapi.plugins['content-manager'].services.contentmanager;

    let entities = [];

    if (ctx.request.query.hasOwnProperty('id_contains')) {
      ctx.request.query.id_eq = ctx.request.query.id_contains;
      delete ctx.request.query.id_contains;
    }

    try {
      if (_.has(ctx.request.query, '_q')) {
        entities = await contentManagerService.search(
          { model },
          ctx.request.query
        );
      } else {
        entities = await contentManagerService.fetchAll(
          { model },
          ctx.request.query
        );
      }
    } catch (e) {
      strapi.log.error(
        'extensions.content-manager.controllers.ContentManager.find',
        e
      );
    }

    ctx.body = entities;
  },
};
```
It will require you to pass whole ID in select, but at least it is usable then.

**Comment by aksdevac** [comment]
Hello, this issue is blocking user from creating entry with relational field (if field is id/_id). in our case there is no other field to uniquely identify the record, so have to use id as relational field in configure the view, if I paste full id manually it keeps showing error. & does not pick up its value, so its very difficult to create entry using admin.

**Comment by viters** [comment]
@aksdevac 
Adjustment for fix in 3.6.5:

Create file `app/extensions/content-manager/controllers/relations.js` with content:

```js
'use strict';

const { has, prop, pick, concat } = require('lodash/fp');
const {
  PUBLISHED_AT_ATTRIBUTE,
} = require('strapi-utils').contentTypes.constants;

const { getService } = require('strapi-plugin-content-manager/utils');

module.exports = {
  async find(ctx) {
    const { model, targetField } = ctx.params;
    const { _component, ...query } = ctx.request.query;
    const { idsToOmit } = ctx.request.body;

    if (!targetField) {
      return ctx.badRequest();
    }

    const modelDef = _component
      ? strapi.db.getModel(_component)
      : strapi.db.getModel(model);

    if (!modelDef) {
      return ctx.notFound('model.notFound');
    }

    const attr = modelDef.attributes[targetField];
    if (!attr) {
      return ctx.badRequest('targetField.invalid');
    }

    const target = strapi.db.getModelByAssoc(attr);

    if (!target) {
      return ctx.notFound('target.notFound');
    }

    if (idsToOmit && Array.isArray(idsToOmit)) {
      query._where = query._where || {};
      query._where.id_nin = concat(query._where.id_nin || [], idsToOmit);
    }

    const entityManager = getService('entity-manager');

    let entities = [];

    if (query.hasOwnProperty('id_contains')) {
      query.id_eq = query.id_contains;
      delete query.id_contains;
    }

    if (has('_q', ctx.request.query)) {
      entities = await entityManager.search(query, target.uid);
    } else {
      entities = await entityManager.find(query, target.uid);
    }

    if (!entities) {
      return ctx.notFound();
    }

    const modelConfig = _component
      ? await getService('components').findConfiguration(modelDef)
      : await getService('content-types').findConfiguration(modelDef);

    const field =
      prop(`metadatas.${targetField}.edit.mainField`, modelConfig) || 'id';
    const pickFields = [field, 'id', target.primaryKey, PUBLISHED_AT_ATTRIBUTE];

    ctx.body = entities.map(pick(pickFields));
  },
};
```

**Comment by derrickmehaffy** [comment]
shouldn't be an issue in the v4, and we no longer support mongo
